src/main/java/store/bookshop/service/category/CategoryServiceImpl.java

```diff
@@ -0,0 +1,66 @@
+package store.bookshop.service.category;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import store.bookshop.dto.book.BookDto;
+import store.bookshop.dto.category.CategoryDto;
+import store.bookshop.dto.category.CreateCategoryRequestDto;
+import store.bookshop.exception.EntityNotFoundException;
+import store.bookshop.mapper.BookMapper;
+import store.bookshop.mapper.CategoryMapper;
+import store.bookshop.model.Category;
+import store.bookshop.repository.BookRepository;
+import store.bookshop.repository.CategoryRepository;
+
+@Service
+@RequiredArgsConstructor
+public class CategoryServiceImpl implements CategoryService {
+    private final CategoryRepository categoryRepository;
+    private final CategoryMapper categoryMapper;
+    private final BookRepository bookRepository;
+    private final BookMapper bookMapper;
+
+    @Override
+    public List<CategoryDto> findAll() {
+        return categoryRepository.findAll().stream()
+                .map(categoryMapper::toDto)
+                .collect(Collectors.toList());
+    }
+
+    @Override
+    public CategoryDto getById(Long id) {
+        Category category = categoryRepository.findById(id).orElseThrow(
+                () -> new EntityNotFoundException("Can't find book by id: " + id));
+        return categoryMapper.toDto(category);
+    }
+
+    @Override
+    public List<BookDto> getBooksByCategoryId(Long id) {
+        return bookRepository.findByCategories_Id(id).stream()
+                .map(bookMapper::toDto)
+                .toList();
+    }
+
+    @Override
+    public CategoryDto save(CreateCategoryRequestDto requestDto) {
+        Category category = categoryMapper.toModel(requestDto);
+        return categoryMapper.toDto(categoryRepository.save(category));
+    }
+
+    @Override
+    public CategoryDto update(Long id, CreateCategoryRequestDto requestDto) {
+        if (categoryRepository.findById(id).isEmpty()) {
```

Comment:
existsById()

Reply:
![image](https://github.com/Paradoxallist/bookshop/assets/106034974/3ddd0fdf-3aac-45e9-9c20-de9bcda76670)
![image](https://github.com/Paradoxallist/bookshop/assets/106034974/726190f5-9765-45b8-9211-089b6de0c4e3)
please agree among yourselves how it should be...